dask/array/tests/test_atop.py

```diff
@@ -31,258 +32,345 @@
+                ),
                 [(a, "ij"), (a, "ji")],
             ),
         ],
+        # Case 6
         [
             [
-                (c, "i", {c: (add, _0, _1)}, [(a, "i"), (b, "i")]),
-                (d, "i", {d: (inc, _0)}, [(c, "i")]),
+                (c, "i", Task(c, add, _0, _1), [(a, "i"), (b, "i")]),
+                (d, "i", Task(d, inc, _0), [(c, "i")]),
             ],
             (
                 d,
                 "i",
-                {d: (inc, _unique_dep(c, "i")), _unique_dep(c, "i"): (add, _0, _1)},
+                Task.fuse(
+                    Task(d, inc, TaskRef(_unique_dep(c, "i"))),
+                    Task(_unique_dep(c, "i"), add, _0, _1),
+                ),
                 [(a, "i"), (b, "i")],
             ),
         ],
+        # Case 7
         [
             [
-                (b, "ij", {b: (np.transpose, _0)}, [(a, "ji")]),
-                (d, "ij", {d: (np.dot, _0, _1)}, [(b, "ik"), (c, "kj")]),
+                (b, "ij", Task(b, np.transpose, _0), [(a, "ji")]),
+                (d, "ij", Task(d, np.dot, _0, _1), [(b, "ik"), (c, "kj")]),
             ],
             (
                 d,
                 "ij",
-                {
-                    d: (np.dot, _unique_dep(b, "ik"), _0),
-                    _unique_dep(b, "ik"): (np.transpose, _1),
-                },
+                Task.fuse(
+                    Task(
+                        d,
+                        np.dot,
+                        TaskRef(_unique_dep(b, "ik")),
+                        _0,
+                    ),
+                    Task(_unique_dep(b, "ik"), np.transpose, _1),
+                ),
                 [(c, "kj"), (a, "ki")],
             ),
         ],
+        # Case 8
         [
             [
-                (c, "i", {c: (add, _0, _1)}, [(a, "i"), (b, "i")]),
-                (f, "i", {f: (add, _0, _1)}, [(d, "i"), (e, "i")]),
-                (g, "i", {g: (add, _0, _1)}, [(c, "i"), (f, "i")]),
+                (c, "i", Task(c, add, _0, _1), [(a, "i"), (b, "i")]),
+                (f, "i", Task(f, add, _0, _1), [(d, "i"), (e, "i")]),
+                (g, "i", Task(g, add, _0, _1), [(c, "i"), (f, "i")]),
             ],
             (
                 g,
                 "i",
-                {
-                    g: (add, _unique_dep(c, "i"), _unique_dep(f, "i")),
-                    _unique_dep(f, "i"): (add, _2, _3),
-                    _unique_dep(c, "i"): (add, _0, _1),
-                },
+                Task.fuse(
+                    Task(
+                        g,
+                        add,
+                        TaskRef(_unique_dep(c, "i")),
+                        TaskRef(_unique_dep(f, "i")),
+                    ),
+                    Task(_unique_dep(f, "i"), add, _2, _3),
+                    Task(_unique_dep(c, "i"), add, _0, _1),
+                ),
                 [(a, i), (b, i), (d, i), (e, i)],
             ),
         ],
```

Comment:
This one appears to be flaky. Haven't determined the source yet but I believe it is not the test but the rewrite logic. Will have to dig into this a little. I also saw flakes in case 9 but much, much more rarely (and another two digit case number I can't remember any longer)

Reply:
this was caused by a tokenization error in task spec and is fixed now